golem-worker-service-base/src/service/api_deployment.rs

```diff
@@ -86,12 +89,22 @@ impl<Namespace: ApiNamespace, ApiDefinition> ApiDeploymentService<Namespace>
                     "Error getting api definition: {}",
                     err
                 ))
-            })
-        {
-            return Err(ApiDeploymentError::ApiDefinitionNotFound(
-                api_definition_key.namespace,
-                api_definition_key.id,
-            ));
+            })?
+            .ok_or(ApiDeploymentError::ApiDefinitionNotFound(
+                api_definition_key.namespace.clone(),
+                api_definition_key.id.clone(),
+            ))?;
+
+        if definition.is_draft() {
```

Comment:
there is existing deployment check,

i think we should update it only in case when it not fail

Reply:
If there is an existing deployment for a draft definition than we have an invalid state and should fail any way.

Reply:
Fixed